### PR TITLE
Warn about long-running transactions

### PR DESCRIPTION
**Pull request by mponomar**

### CODE REVIEW

File: berkdb/build/db.h [hunk 1]
```diff
@@ -1060,6 +1061,7 @@ struct __db_txn_active {
 	DB_LSN	  lsn;			/* LSN when transaction began */
 	u_int32_t xa_status;		/* XA status */
 	u_int8_t  xid[DB_XIDDATASIZE];	/* XA global transaction ID */
+    time_t start_time;
```
Comment (akshatsikarwar) [inline berkdb/build/db.h]: Tab

File: berkdb/dbinc/txn.h [hunk 2]
```diff
@@ -68,6 +68,7 @@ typedef struct __txn_detail {
 	u_int32_t bqual;		/* bqual_length from XID */
 	u_int32_t gtrid;		/* gtrid_length from XID */
 	int32_t format;			/* XA format */
+    time_t start_time;
 } TXN_DETAIL;
```
Comment (akshatsikarwar) [inline berkdb/dbinc/txn.h]: Tab

File: berkdb/txn/txn.c [hunk 3]
```diff
@@ -597,6 +597,8 @@ __txn_begin_int_int(txn, internal, retries, we_start_at_this_lsn)
 		MUTEX_THREAD_UNLOCK(dbenv, mgr->mutexp);
 	}
 
+    td->start_time = time(NULL);
```
Comment (akshatsikarwar) [inline berkdb/txn/txn.c]: Tab

**Review by akshatsikarwar — DISMISSED** [review]
I was going to do it in the lock manager and detect there. But this is much better.

File: tests/oldtran.test/runit [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+dbname=$1
+if [[ -z $dbname ]] ; then
+    echo dbname missing
+    exit 1
+fi
+
+cdb2sql $dbname "exec procedure sys.cmd.send('oldest_transaction_alarm_threshold 10')"
+cdb2sql $dbname "exec procedure sys.cmd.send('bdb leaktran')"
+
+sleep 20
```
Comment (nirbhayc) [inline tests/oldtran.test/runit]: Consider reducing the sleep amount (alarm threshold 1sec, sleep 2?).

File: tests/oldtran.test/Makefile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+ifeq ($(TESTSROOTDIR),)
+  include ../testcase.mk
+else
+  include $(TESTSROOTDIR)/testcase.mk
+endif
```
Comment (markhannum) [inline tests/oldtran.test/Makefile]: Maybe make this
ifeq ($(TEST_TIMEOUT),)
   export TEST_TIMEOUT=1m
endif

**Comment by akshatsikarwar** [comment]
This catches a long rebuild?

**Pull request closed without merging**